automation/load/src/gatling/kotlin/uk/co/baconi/oauth/api/gatling/endpoints/AuthorisationEndpoint.kt

```diff
@@ -30,10 +32,14 @@ object AuthorisationEndpoint {
     }
 
     object Transforms {
+
+        private val URI.queryParams: Map<String, String>
+            get() = query.split("&").map { it.split("=", limit = 2) }.associate { it[0] to it[1] }
+
         val headerToAuthorisationCode: (String?) -> String? = { header ->
             when (header) {
                 null -> null
-                else -> Uri.create(header).encodedQueryParams.find { param -> param.name == "code" }?.value
```

Comment:
Gatling's `Uri` class doesn't support some custom protocols used when doing native call backs for public clients.